### PR TITLE
add a restart command and auto-run state.py when bot.py is run

### PR DESCRIPTION
**Pull request by ovkulkarni**
When running `./bot.py`, `state.py` is automatically called from `bot.py`, and the restart command terminates `state.py` and restarts `bot.py` from the beginning, causing `state.py` to be run again.

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -4,12 +4,16 @@
...
+import os
+
+state = sp.Popen(['python','state.py'])
 
 conf_file = sys.argv[1] if len(sys.argv) > 1 else "config.yml"
 with open(conf_file) as f:
     config = yaml.load(f)
 
-admins = ["fwilson", "sdrodge"]
+admins = ["fwilson", "sdrodge", "okulkarni"]
```
Comment (an-empty-string) [inline bot.py:17]: oh please

File: bot.py [hunk 2]
```diff
@@ -4,12 +4,16 @@
 import threading
 import sys
 import yaml
+import subprocess as sp
+import os
+
+state = sp.Popen(['python','state.py'])
```
Comment (an-empty-string) [inline bot.py]: make whether or not to do this a configuration option, defaulting to false

File: bot.py [hunk 3]
```diff
@@ -66,6 +70,9 @@ def on_message(self, message, nickname, channel, is_query):
                 args = message.split()
                 denied[args[1]] = []
                 bot.send(channel, "ok!")
+            elif message == '!restart':
+                sp.Popen.terminate(state)
+                os.execl(sys.executable, sys.executable, *sys.argv)
```
Comment (an-empty-string) [inline bot.py]: what are the implications of this? will the existing irc connection stay open?

File: bot.py (hunk 1 above)
Comment (ovkulkarni) [inline bot.py:17]: yeah, that was for when i was running it, forgot to change

File: bot.py (hunk 3 above)
Comment (ovkulkarni) [inline bot.py]: No, it terminates state, ends the irc connection, and then starts over

File: bot.py (hunk 2 above)
Comment (ovkulkarni) [inline bot.py]: ok

**Comment by ovkulkarni** [comment]
bump @fwilson42

**Pull request merged**